Sample audio file to stream convertor

jayawin opened the pull request:
Sample audio file to stream convertor classes
There are two classes, to show 2 different ways to convert.

**AudioFileToStream:**
    This class is responsible for converting an audio file to a stream.
    The input file is loaded into an AudioSegment and then converted into a stream.

**AudioFileToStreamPyAudio**
This class is responsible for converting an audio file in wav format to a stream using PyAudio.

aponcedeleonch reviewed (approved):
Looks ok, couple of suggestions:
1. Could you please type hint the variables you use. For example in `def __init__(self, file, chunk=1024)`. That way is easier to enforce typechecking and also making the code understandable
2. There are a couple of linting issues

jayawin commented:
My intention was not to merge it, I just wanted to check if this is the
right direction.
Sorry for the trouble, but I will add the linting and typechecks

On Mon, Feb 3, 2025 at 8:29 PM Alejandro Ponce de Leon <
***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Looks ok, couple of suggestions:
>
>    1. Could you please type hint the variables you use. For example in def
>    __init__(self, file, chunk=1024). That way is easier to enforce
>    typechecking and also making the code understandable
>    2. There are a couple of linting issues
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aponcedeleonch/nite/pull/2#pullrequestreview-2590699521>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOKOIWHT5DRKDSRI56DR74L2N6YPDAVCNFSM6AAAAABWMVZBWSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKOJQGY4TSNJSGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jayawin commented:
Addressed the comments. Please feel free to comment, if it has to improved.
I added try except block. You haven't used else where, so if that has to be removed, let me know
